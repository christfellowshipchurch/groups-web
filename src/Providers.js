import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
