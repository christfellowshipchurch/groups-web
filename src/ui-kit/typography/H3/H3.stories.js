import React from 'react';
import { storiesOf } from '@storybook/react';

import H3 from '.';

storiesOf('ui-kit/typography/H3', module)
  .add('default', () => (
    <H3>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H3>
  ))
  .add('isLoading', () => (
    <H3 isLoading showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H3>
  ))
  .add('padded', () => <H3 padded>Padded Heading Big and Long</H3>)
  .add('padded + isLoading', () => (
    <H3 isLoading padded showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H3>
  ));
