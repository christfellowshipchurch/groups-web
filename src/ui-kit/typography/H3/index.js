import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { Line } from '../../Placeholder';

const verticalRhythm = css`
  ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(
          1.125
        )} 0 ${theme.helpers.verticalRhythm(0.75)} 0`
      : 'initial'};
`;

const StyledH3 = styled.h3.attrs({
  className: 'ui-kit-typography-H3-StyledH3',
})`
  font-size: 1.5rem;
  font-weight: 900;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(1.5, 1.15)};
  margin: ${verticalRhythm};

  /* Styles passed via the style prop */
  ${({ customStyles }) => customStyles}
`;

const Placeholder = styled(Line).attrs({
  className: 'ui-kit-H1-Placeholder',
})`
  height: 1.5rem; /* Should match H3 font-size */
  margin: ${({ padded, theme }) =>
    // if padded...
    padded
      ? // return rhythm so placeholder matches text position to prevent jitter/jank
        verticalRhythm
      : // else default to...
        `calc(${theme.helpers.verticalRhythm(
          1.5,
          1.15
        )} - 1.5rem) 0`}; /* line-height - font-size = space between lines */
`;

// We pass `style` as a prop to `StyledH3` so that custom styles are handled by styled components and not react (inline style)
const H3 = ({ children, isLoading, padded, style, ...props }) => (
  <Placeholder isLoading={isLoading} padded={padded} showLoadingAnimation>
    <StyledH3 padded={padded} customStyles={style} {...props}>
      {children}
    </StyledH3>
  </Placeholder>
);

H3.defaultProps = {
  isLoading: false,
};

H3.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
  style: PropTypes.any, // eslint-disable-line
};

H3.displayName = 'ui-kit.typography.H3';

export default H3;
