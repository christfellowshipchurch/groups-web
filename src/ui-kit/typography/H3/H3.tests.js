import React from 'react';
import renderer from 'react-test-renderer';

import Providers from '../../../Providers';

import H3 from '.';

describe('the H3 component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <H3>Default H3 text</H3>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H3 style={salmonText}>Salmon text</H3>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H3 padded>H3 text with vertical rhythm styled via margin.</H3>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <H3 isLoading>Default H3 text</H3>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
