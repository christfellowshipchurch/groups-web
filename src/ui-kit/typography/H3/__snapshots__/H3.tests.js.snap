// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H3 component should accept and render passed in styles 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: 900;
  line-height: 1.7249999999999999rem;
  margin: initial;
  color: salmon;
}

<h3
  className="c0 ui-kit-typography-H3-StyledH3"
>
  Salmon text
</h3>
`;

exports[`the H3 component should render 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: 900;
  line-height: 1.7249999999999999rem;
  margin: initial;
}

<h3
  className="c0 ui-kit-typography-H3-StyledH3"
>
  Default H3 text
</h3>
`;

exports[`the H3 component should render a loading state 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1.5rem;
  margin: calc(1.7249999999999999rem - 1.5rem) 0;
}

<mark
  className="c0 show-loading-animation c1 ui-kit-H1-Placeholder ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`the H3 component should render a loading state with vertical rhythm (padded) 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1.5rem;
  margin: 1.6875rem 0 1.125rem 0;
}

<mark
  className="c0 show-loading-animation c1 ui-kit-H1-Placeholder ui-kit.Placeholder.Line custom-placeholder"
  width="100%"
/>
`;

exports[`the H3 component should render with vertical rhythm (padded) 1`] = `
.c0 {
  font-size: 1.5rem;
  font-weight: 900;
  line-height: 1.7249999999999999rem;
  margin: 1.6875rem 0 1.125rem 0;
}

<h3
  className="c0 ui-kit-typography-H3-StyledH3"
>
  H3 text with vertical rhythm styled via margin.
</h3>
`;
